src/Symfony/Component/HttpKernel/Controller/ArgumentResolver/RequestPayloadValueResolver.php

```diff
@@ -114,19 +114,19 @@ public function resolve(Request $request, ArgumentMetadata $argument): iterable
         return [];
     }
 
-    private function mapQueryString(Request $request, string $type, MapQueryString $attribute): ?object
+    private function mapQueryString(Request $request, ArgumentMetadata $argument, MapQueryString $attribute): ?object
     {
         if (!$data = $request->query->all()) {
             return null;
         }
 
-        return $this->serializer->denormalize($data, $type, null, self::CONTEXT_DENORMALIZE + $attribute->context);
+        return $this->serializer->denormalize($data, $argument->getType(), null, self::CONTEXT_DENORMALIZE + $attribute->context);
     }
 
-    private function mapRequestPayload(Request $request, string $type, MapRequestPayload $attribute): ?object
+    private function mapRequestPayload(Request $request, ArgumentMetadata $argument, MapRequestPayload $attribute): ?object
```

Comment:
seems irrelevant

Reply:
The new `mapRequestPayload()` implementation requires `ArgumentMetadata` as an argument,
the call happens in a dynamic way (search for `$this->$payloadMapper()` in the `resolve()` method
so this change is required as the caller sends a new value

Reply:
I also don't get this change: where do we use $argument and not only its type?

Reply:
Fixed. Thanks for insisting on that, I wasn't seeing the problem.